dockerfiles/Dockerfile-build-crypt

```diff
@@ -0,0 +1,145 @@
+# Build libsodium
+FROM emscripten/emsdk:latest AS SODIUM
+
+RUN wget https://download.libsodium.org/libsodium/releases/libsodium-1.0.18-stable.tar.gz \
+    && tar xvf libsodium-1.0.18-stable.tar.gz && cd libsodium-stable \
+    && dist-build/emscripten.sh --sumo
+
+# Build openssl
+FROM emscripten/emsdk:latest AS OPENSSL
+
+RUN wget https://www.openssl.org/source/openssl-1.1.1l.tar.gz \
+    && tar xvf openssl-1.1.1l.tar.gz \
+    && cd openssl-1.1.1l \
+    && emconfigure ./Configure linux-generic64 no-asm no-threads no-engine no-hw no-weak-ssl-ciphers no-dtls no-shared no-dso --prefix=/emsdk/upstream \
+    && sed -i 's|^CROSS_COMPILE.*$|CROSS_COMPILE=|g' Makefile \
+    && sed -i '/^CFLAGS/ s/$/ -D__STDC_NO_ATOMICS__=1/' Makefile \
+    && sed -i '/^CXXFLAGS/ s/$/ -D__STDC_NO_ATOMICS__=1/' Makefile \
+    && emmake make -j 2 all \
+    && emmake make install
+
+# Build libcrypt4gh
+FROM emscripten/emsdk:latest AS LIBCRYPT4GH
+
+COPY --from=SODIUM /src/libsodium-stable/libsodium-js-sumo/include/ /emsdk/upstream/include/
+COPY --from=SODIUM /src/libsodium-stable/libsodium-js-sumo/lib/ /emsdk/upstream/lib/
+
+RUN sudo apt-get update \
+    && sudo apt-get upgrade -y \
+    && sudo apt-get install -y autoconf
+
+RUN git clone https://github.com/silverdaz/libcrypt4gh.git
```

Comment:
we might want to fork this to our CSCfi in case changes happen that are out of our control

Reply:
https://github.com/cscfi/libcrypt4gh